File: frontends/Julia/Gencode/sympyassign2.jl
function sympyassign2(mystr::String, f, udg, wdg, uhg)

    f = f[:]
    str1 = getccode(f, "f[")
    str1 = "\t\t{\n" * str1 * "\t\t}\n"
    mystr = mystr * str1

    nf = length(f);    
            
    if !isnothing(udg) && !isempty(udg)
        nu = length(udg);
        f_udg = [SymPy.symbols("f_udg$i") for i=1:(nf*nu)];
        for n = 1:nu
          for m = 1:nf      
            f_udg[m + nf*(n-1)] = diff(f[m],udg[n]);      
          end
        end        
        if !isempty(f_udg)
            str2 = getccode(f_udg, "f_udg[")
            str2 = "\t\t{\n" * str2 * "\t\t}\n"
            mystr = mystr * str2
        end
    end

    if !isnothing(wdg) && !isempty(wdg)
        nw = length(wdg);
        f_wdg = [SymPy.symbols("f_wdg$i") for i=1:(nf*nu)];
        for n = 1:nw
          for m = 1:nf      
            f_wdg[m + nf*(n-1)] = diff(f[m],wdg[n]);      
          end
        end                
        if !isempty(f_wdg)
            str3 = getccode(f_wdg, "f_wdg[")
            str3 = "\t\t{\n" * str3 * "\t\t}\n"
            mystr = mystr * str3
        end
    end

    if !isnothing(uhg) && !isempty(uhg)
        nu = length(uhg);
        f_uhg = [SymPy.symbols("f_uhg$i") for i=1:(nf*nu)];
        for n = 1:nu
          for m = 1:nf      
            f_uhg[m + nf*(n-1)] = diff(f[m],uhg[n]);      
          end
        end              
        if !isempty(f_uhg)
            str4 = getccode(f_uhg, "f_uhg[")
            str4 = "\t\t{\n" * str4 * "\t\t}\n"
            mystr = mystr * str4
        end
    end

    return mystr
end

File: frontends/Python/Gencode/sympyassign2.py
import sympy
from getccode import getccode

def sympyassign2(mystr, f, udg, wdg, uhg):
    # Generate C code for the original function
    str1 = getccode(f, "f")
    str1 = "\t\t{\n" + str1 + "\t\t}\n"
    mystr += str1

    nf = len(f)

    # Process udg if not None and not empty
    if udg is not None and len(udg) > 0:
        nu = len(udg)
        f_udg = [sympy.symbols(f"f_udg{i}") for i in range(nf * nu)]
        for n in range(nu):
            for m in range(nf):
                f_udg[m + nf * n] = sympy.diff(f[m], udg[n])
        if len(f_udg) > 0:
            str2 = getccode(f_udg, "f_udg")
            str2 = "\t\t{\n" + str2 + "\t\t}\n"
            mystr += str2

    # Process wdg if not None and not empty
    if wdg is not None and len(wdg) > 0:
        nw = len(wdg)
        f_wdg = [sympy.symbols(f"f_wdg{i}") for i in range(nf * nw)]
        for n in range(nw):
            for m in range(nf):
                f_wdg[m + nf * n] = sympy.diff(f[m], wdg[n])
        if len(f_wdg) > 0:
            str3 = getccode(f_wdg, "f_wdg")
            str3 = "\t\t{\n" + str3 + "\t\t}\n"
            mystr += str3

    # Process uhg if not None and not empty
    if uhg is not None and len(uhg) > 0:
        nu = len(uhg)
        f_uhg = [sympy.symbols(f"f_uhg{i}") for i in range(nf * nu)]
        for n in range(nu):
            for m in range(nf):
                f_uhg[m + nf * n] = sympy.diff(f[m], uhg[n])
        if len(f_uhg) > 0:
            str4 = getccode(f_uhg, "f_uhg")
            str4 = "\t\t{\n" + str4 + "\t\t}\n"
            mystr += str4

    return mystr

# Example usage:
# f = sympy.symbols('f0:10')
# udg = sympy.symbols('udg0:10')
# wdg = sympy.symbols('wdg0:10')
# uhg = sympy.symbols('uhg0:10')
# mystr = ""
# result = sympyassign2(mystr, f, udg, wdg, uhg)
# print(result)
